### PR TITLE
Test method for starting gRPC server temp socket

### PR DESCRIPTION
**Pull request by azdagron**
Also fixes the handling of the error returned when the server is stopped to prevent a false-negative test failure.

### CODE REVIEW

**Review by amartinezfayo — APPROVED** [review]
LGTM!

File: test/spiretest/socketapi.go [hunk 1]
```diff
@@ -60,6 +67,11 @@ func StartGRPCSocketServer(t *testing.T, socketPath string, registerFn func(s *g
 	}()
 	t.Cleanup(func() {
 		server.Stop()
-		require.NoError(t, <-errCh)
+		err := <-errCh
+		switch {
```
Comment (APTy) [inline test/spiretest/socketapi.go:71]: nit:
```
switch err {
case nil, grpc.ErrServerStopped:
default:
    t.Fatal(err)
}
```
i'd even prefer putting a `return` in the `case` statement to be more explicit about what's happening, but up to you

**Review by APTy — APPROVED** [review]
LGTM. Are you planning to use `StartGRPCSocketServerOnTempSocket()` in a future PR I'm guessing? Wondering why it's not used as part of this PR

**Comment by azdagron** [comment]
Yup, it's for future test work!

**Pull request merged**